ksql-engine/src/main/java/io/confluent/ksql/services/DefaultServiceContext.java

```diff
@@ -29,38 +30,87 @@
 public class DefaultServiceContext implements ServiceContext {
 
   private final KafkaClientSupplier kafkaClientSupplier;
-  private final Admin adminClient;
-  private final KafkaTopicClient topicClient;
-  private final Supplier<SchemaRegistryClient> srClientFactory;
+  private final Supplier<Admin> adminClientSupplier;
+  private final Supplier<KafkaTopicClient>  topicClientSupplier;
+  private final Supplier<SchemaRegistryClient> srClientSupplier;
   private final SchemaRegistryClient srClient;
```

Comment:
why do we need this field also, if there is a supplier already?

Reply:
I created an issue #3852 and added the link to the description of this PR